File: __OTHER_FRAMEWORKS__/grafana/dashboards-as-code-workshop-main/reference/part-one-java/src/main/java/lab/grafana/Manifest.java
package lab.grafana;

import com.grafana.foundation.dashboard.Dashboard;
import com.grafana.foundation.resource.Metadata;

import java.util.Map;

public class Manifest {
    public static com.grafana.foundation.resource.Manifest dashboard(String folderUid, Dashboard dashboard) {
        Metadata meta = new Metadata();
        meta.name = dashboard.uid;
        meta.annotations = Map.of(
                "grafana.app/folder", folderUid
        );

        return new com.grafana.foundation.resource.Manifest(
                "dashboard.grafana.app/v1beta1",
                "Dashboard",
                meta,
                dashboard
        );
    }
}

File: __OTHER_FRAMEWORKS__/grafana/dashboards-as-code-workshop-main/reference/part-two-php/src/Grafana/Manifest.php
<?php

namespace App\Grafana;

use Grafana\Foundation\Dashboard\Dashboard;
use Grafana\Foundation\Resource;

class Manifest
{
    public static function dashboard(string $folderUid, Dashboard $dashboard): Resource\Manifest
    {
        return new Resource\Manifest(
            apiVersion: 'dashboard.grafana.app/v1beta1',
            kind: 'Dashboard',
            metadata: new Resource\Metadata(
                annotations: [
                    "grafana.app/folder" => $folderUid,
                ],
                name: $dashboard->uid,
            ),
            spec: $dashboard,
        );
    }
}
